WordPress/src/main/java/org/wordpress/android/ui/main/MySiteFragment.java

```diff
@@ -73,126 +78,110 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
                 ActivityLauncher.viewCurrentBlogComments(getActivity());
             }
         });
 
-        WPTextView themesTextView = (WPTextView) rootView.findViewById(R.id.my_site_themes_text_view);
-        themesTextView.setOnClickListener(new View.OnClickListener() {
+        mThemesContainer.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
                 ActivityLauncher.viewCurrentBlogThemes(getActivity());
             }
         });
 
-        WPTextView settingsTextView = (WPTextView) rootView.findViewById(R.id.my_site_settings_text_view);
-        settingsTextView.setOnClickListener(new View.OnClickListener() {
+        rootView.findViewById(R.id.row_settings).setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
                 ActivityLauncher.viewBlogSettingsForResult(getActivity(), mBlog);
             }
         });
 
-        WPTextView viewAdminTextView = (WPTextView) rootView.findViewById(R.id.my_site_view_admin_text_view);
-        viewAdminTextView.setOnClickListener(new View.OnClickListener() {
+        rootView.findViewById(R.id.row_admin).setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
                 ActivityLauncher.viewBlogAdmin(getActivity(), mBlog);
             }
         });
 
-        LinearLayout addPostContainer = (LinearLayout) rootView.findViewById(R.id.my_site_posts_add_button_container);
-        addPostContainer.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                ActivityLauncher.addNewBlogPostOrPage(getActivity(), mBlog, false);
-            }
-        });
-
-        LinearLayout addPageContainer = (LinearLayout) rootView.findViewById(R.id.my_site_pages_add_button_container);
-        addPageContainer.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                ActivityLauncher.addNewBlogPostOrPage(getActivity(), mBlog, true);
-            }
-        });
-
-        LinearLayout addMediaContainer = (LinearLayout) rootView.findViewById(R.id.my_site_media_add_button_container);
-        addMediaContainer.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                ActivityLauncher.addMedia(getActivity());
-            }
-        });
-
         refreshBlogDetails();
 
         return rootView;
     }
 
+    private void showSitePicker() {
+        if (isAdded()) {
+            ReaderAnim.showFab(mFabView, false);
+            int localBlogId = (mBlog != null ? mBlog.getLocalTableBlogId() : 0);
+            ActivityLauncher.showSitePickerForResult(getActivity(), localBlogId);
+        }
+    }
+
     @Override
     public void onActivityResult(int requestCode, int resultCode, Intent data) {
```

Comment:
I noticed `onActivityResult` is not called when "Don't keep activities" is checked in dev options (because the fragment it null here: https://github.com/wordpress-mobile/WordPress-Android/blob/feature/my-site-fab/WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java#L286)